### DISCUSSION TITLE
Possible improvements to Pagination component

### DISCUSSION THREAD

**Discussion by daniel-ac-martin** [UKHomeOffice/design-system#376] [Ideas]
From @haggishunt56:

1) Replace items prop with nested <TimelineEvent> elements
2) Add buttons and links to timeline items (TBC)
3) Add support for collapsable items, to allow users to display long timelines concisely

**Comment by guermanc** [comment]
## Accessibility considerations
We currently advise that "large clickable hit areas are provided and are clearly spaced out" for the pagination component. 

The current implementation means that the hit areas are below those recommended by the AAA WCAG Success Criterion 2.5.5 Target Size, which are 44 x 44 CSS pixels. There are also recommendations for large touch target sizes in [Apple's Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/foundations/accessibility/#best-practices) and [Google's Material Design](https://m3.material.io/foundations/accessible-design/accessibility-basics).

When reducing the viewport, overlap of touch targets can also be observed.

This is a AAA criteria of course, but I feel there is scope to improve the current implementation. GOV.UK seems to have a [better implementation](https://design-system.service.gov.uk/components/pagination/) with touch targets meeting the 44 x 44 CSS pixel criteria as well as reflowing correctly.

**Comment by krisztin** [comment]
Hi 👋️, Accessibility/UX Dev here. Spotted a few issues with your pagination component:

## `<nav>`

- The attribute role="navigation" is not needed on a native <nav> element
- The `<nav>` is already announced as "navigation" therefore the "navigation" in the aria-label is redundant. It could simply be aria-label="Pagination"

## Current page

- The aria-current is set to `"true"` which is then announced as "link current" by screen readers. `"page"` would be a more appropriate value resulting in "page current".
- This element also has an aria-label that does not add additional information, just repeats everything (`aria-current` will let users know it's the current page and the `<a>` inner text of the page's number would communicate the number)

## Pages
`aria-labels` are a bit too verbose. Putting "go to" into each one makes the screen reader experience unnecessarily protracted. People already expect a link to "go to" somewhere.

## Arrows
These only need an `aria-hidden="true"` attribute.
[The presentation role is used to "erase" the default behaviour of an element](https://www.w3.org/WAI/ARIA/apg/practices/hiding-semantics/) such as if you put presentation on an h1 it becomes a div for screen readers.

**Comment by willrichards** [comment]
Thank you @krisztin for providing this feedback.

The design system working group have discussed this component and our approach is to move towards the GOV UK pagination component instead of this one. An issue has been created to get this updated: [https://github.com/UKHomeOffice/design-system/issues/651](https://github.com/UKHomeOffice/design-system/issues/651)

**Comment by Jack-Hewson** [comment]
Can we please also include a pagination component similar to the one listed [here](https://design-system.service.gov.uk/components/pagination/ ) under "For navigating between pages of items"

The benefit of this component is that it allows you to jump to the start or the end of a range of pages without needing to click "Next" or "Previous" multiple times
It's also very beneficial if you do not include the page number in the URL and this is the only way to get to a page

**Comment by Jack-Hewson** [comment]
Can we please also include pagination in the style of the component found [here](https://design-system.service.gov.uk/components/pagination/) under "For navigating between pages of items"

This is useful, compared to the current pagination component, because it allows a user to jump to end pages or start pages in fewer clicks. This can be particularly beneficial in circumstances where pagination isn't included within the URL
